src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Container, Row, Col, Spinner,
+} from 'react-bootstrap';
+import Form from 'react-bootstrap/Form';
+import { checkValidity, isFormValidCheck, checkError } from '../../utils/validation';
+import './SignUp.css';
+
+class SignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: {
+        value: '',
+        valid: false,
+        validation: {
+          required: true,
+          email: true,
+        },
+        touched: false,
+      },
+      password: {
+        value: '',
+        valid: false,
+        validation: {
+          required: true,
+          minLength: 6,
+        },
+        touched: false,
+      },
+      confirm: {
+        value: '',
+        valid: false,
+        validation: {
+          required: true,
+          match: 'password',
+        },
+        touched: false,
+      },
+      role: {
+        value: '',
+        valid: false,
+        validation: {
+          required: true,
+        },
+        touched: false,
+      },
+      terms: {
+        value: false,
+        valid: false,
+        touched: false,
+      },
+      isFormValid: false,
+      loading: false,
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.setState({ loading: true });
+  }
+
+  handleInputChange(e) {
+    const name = e.target;
+    const value = e.target;
```

Comment:
Welldone @Yasholma , i wanted to try typing on the input fields to see how the validation works but the input fields were not updating with the values until i changed this line of code to `const { name } = e.target` and `const {value} = e.target` did you leave it out intentionally?, and  i think you can add `e.preventDefault()` also at the top of the `handleInputChange` method.

Reply:
I destructured that initially, then codacy slapped me. I tried this current code it was working, maybe my browser cached it. But I repushed a new version again

Reply:
Nice